cli/tests/setup_suite.bash

```diff
@@ -420,6 +442,7 @@ home_setup() {
   fi
   xdg_tmp_setup
   flox_vars_setup
+  user_dotfiles_setup
```

Comment:
I thought about calling that the "nightmare_user_dotfiles_setup" function, because the idea here is bathe our entire test suite in the harshest environment imaginable, and thanks to Apple and nix-darwin we have some pretty good ideas of what that might look like!  😄  I thought about applying the environment only to the activate tests, but then thought it would be useful for the entirety of flox; for instance, that's how I discovered that we were relying upon the PATH for pkgdb's use of Nix libraries to find `nix` in its path. This kind of thing is extremely important for ensuring consistency across all Linux variants.

Reply:
Bottom line is we cannot escape the PATH blatting that happens on account of the use of nix-darwin on our Apple build hosts, and that caused test results to be inconsistent between our Linux and Apple hosts. Adding these "nightmare" dotfiles for use on all systems goes a long way towards ensuring the consistency across system types, OS distros, releases, etc. and as long as they increase the likelihood that problems on _any_ system will show up on _all_ systems then that's got to make it more likely that we'll spot problems earlier, and ideally on the mix of linux/darwin developer desktops before it even lands in CI.

I don't doubt that there _could_ be a bug that only manifests itself in a "clean room" configuration, but we have empirical evidence to show that's not what we're encountering today, and that we are in fact encountering the opposite, so that I think is the case for adding them now.

@mkenigs I'm also interested in your thoughts, and in drawing this conversation to a close. Thanks.

Reply:
Will move this to activate.bats so it can be present for all the tests in that suite only.

Reply:
Fixed